streams/src/test/java/org/apache/kafka/streams/integration/StatelessIntegrationTest.java

```diff
@@ -0,0 +1,107 @@
+
+        broker.createTopic(inputTopic, 2, 1);
+        broker.createTopic(outputTopic, 2, 1);
+
+        final String path = TestUtils.tempDirectory(applicationId).getPath();
+
+        final Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, applicationId);
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, broker.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, path);
+        streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 1000);
+        streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        streamsConfiguration.put(IntegrationTestUtils.INTERNAL_LEAVE_GROUP_ON_CLOSE, true);
+
+        final StreamsBuilder builder = new StreamsBuilder();
+        final KStream<String, String> input = builder.stream(inputTopic, Consumed.with(Serdes.String(), Serdes.String()));
+        final KStream<String, String> filter = input.filter((k, s) -> s.length() % 2 == 0);
+        final KStream<String, String> map = filter.map((k, v) -> new KeyValue<>(k, k + v));
+        map.to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
+
+        final Topology topology = builder.build();
+        final KafkaStreams kafkaStreams = new KafkaStreams(topology, streamsConfiguration);
+        try {
+            kafkaStreams.start();
+
+            final Properties producerConfig = new Properties();
+            producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, Serdes.String().serializer().getClass());
+            producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, Serdes.String().serializer().getClass());
+            producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, broker.bootstrapServers());
+            produceKeyValuesSynchronously(
+                inputTopic,
+                asList(new KeyValue<>("a", "b"), new KeyValue<>("c", "de")), producerConfig,
+                new SystemTime()
+            );
+
+            final Properties consumerConfig = new Properties();
+            consumerConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, Serdes.String().deserializer().getClass());
+            consumerConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, Serdes.String().deserializer().getClass());
+            consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, broker.bootstrapServers());
+            consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, "mygroup");
+
+            waitUntilMinRecordsReceived(
+                consumerConfig,
+                outputTopic,
+                1
+            );
+
+            if (new File(path).exists()) {
+                final List<Path> collect =
+                    Files.find(Paths.get(path), 999, (p, bfa) -> true).collect(Collectors.toList());
```

Comment:
Hey @enothereska , thanks for the review!

Just to make sure I follow, you're saying that we should have a test that verifies that Streams actually respects its state directory config?

I agree that it would be a serious bug if it did not, so if we're lacking test coverage on that, I'd be happy to add it.